src/server/DeviceServer_v2.js

```diff
@@ -89,15 +84,15 @@ class DeviceServer {
               );
             }
 
-            this._devicesById.set(deviceID, core);
+            this._devicesById.set(deviceID, device);
             const existingAttributes =
               await this._deviceAttributeRepository.getById(deviceID);
             const deviceAttributes = {
               ...existingAttributes,
-              deviceID: deviceID,
-              ip: core.getRemoteIPAddress(),
-              particleProductId: core._particleProductId,
-              productFirmwareVersion: core._productFirmwareVersion,
+              deviceID,
+              ip: device.getRemoteIPAddress(),
+              particleProductId: device._particleProductId,
```

Comment:
I would eventually like to clean this code up as well. It's bad that we are accessing private properties.